### ISSUE TITLE
Conflict warning when swapping Function App slot

### ISSUE THREAD

**Issue by dgard1981** [microsoft/azure-pipelines-tasks#12354]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Question

**Enter Task Name**: AzureAppServiceManageV0

## Environment
- Server - Azure Pipelines
- Agent - Hosted 

## Issue Description

When using the Azure App Service Manage task to swap slots on a Function App, every run results in a warning -

> ##[warning]Error: Failed to update deployment history. Conflict (CODE: 409)

I found a [slightly different issue](https://github.com/Microsoft/azure-pipelines-tasks/issues/4457) that suggested manually starting the slot before the swap might resolve errors like this, but doing so has not helped.

The slot swap does seem to have taken place, but obviously the question is why am I seeing this error, and how can I get rid of it (so that my release is nice and clean)?

### Task logs

Verbose logs for the task...
[tasklog_6.zip](https://github.com/microsoft/azure-pipelines-tasks/files/4200083/tasklog_6.zip)

**Comment by SumiranAgg** [comment]
@dgard1981 After swap slots operation, we update the deployment history for source and target app service slots. We just add some metadata info for operation details. The warning is generated when these calls fail. Hence the required functionality is complete but this call is failing. Is is possible that swap is not complete by the time we attempt the new calls.

We will check and make these calls more reliable.

**Comment by APIWT** [comment]
We are getting this warning too :) Not a blocker, but it would be awesome if this could be addressed.

**Comment by mkowalskigps** [comment]
Also seeing this in our deployments - seems like things are working correctly though 🤞

**Comment by vadim82** [comment]
We are seeing the same thing. The sideeffect of this is that we are not seeing this deployment in the deployment center of the app service.

**Comment by dgard1981** [comment]
We stopped seeing these issues now.

Our Function Apps and slots are created through an ARM template, and as part of our deployment we were setting App Settings in our slot, before swapping in to production. This meant that we we're setting the value of `WEBSITE_CONTENTSHARE` to be the same in both the slot and production.

It's a pain that this is managed through App Settings, but with the following process prior to setting App Setting values the issue has been resolved -

- Get the current App Settings and check for a value for `WEBSITE_CONTENTSHARE`.
- If a value already exists, use that.
- If a value does not exist, generate a new one - max 32 characters, unique within the Function App.
